### PR TITLE
feat: Allow passing custom attributes for script tags

### PR DESCRIPTION
**Pull request by torotil**
The feature is limited to non-aggregated (preprocess=0) local and external files.

### CODE REVIEW

File: theming.module [hunk 1]
```diff
@@ -31,3 +31,35 @@ function theming_theme_registry_alter(&$hooks) {
...
+ */
+function theming_pre_render_scripts(array $elements) {
+  $rendered_elements = drupal_pre_render_scripts($elements);
+  foreach ($elements['#items'] as $key => $script) {
+    // Local and external scripts keep their path/URL as key, inline scripts and settings don’t.
+    if (is_numeric($key) || $key === 'settings') {
+      continue;
+    }
+    $custom_attributes = $script['attributes'] ?? [];
+    foreach (element_children($rendered_elements['scripts']) as $index) {
```
Comment (katzmo) [inline theming.module]: Wouldn't it be easier to make this the main loop and look for `$elements['#items'][$attributes['src']]['attributes']`? :thinking:

Comment (torotil) [reply theming.module]: True. I’ve turned it around in a new commit and also added a simple test.

File: tests/ModuleTest.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/**
+ * Test hook and callback implementations for the theming module.
+ */
+class ModuleTest extends DrupalUnitTestCase {
+
+  /**
+   * Test that it’s possible to override the type attribute of a <script> tag.
+   */
+  public function testPreRenderScriptsOverridesType() {
+    $example_url = 'https://example.com/test.css';
```
Comment (katzmo) [inline tests/ModuleTest.php]: Lets perhaps make this `test.js` for a more realistic example? (css files should not be rendered as script tags) :wink:

File: theming.module (hunk 1 above)
Comment (katzmo) [reply theming.module]: Beautiful, thanks!

File: tests/ModuleTest.php (hunk 2 above)
Comment (torotil) [reply tests/ModuleTest.php]: Done. :heavy_check_mark:

**Review by katzmo — APPROVED** [review]

**Pull request merged**